File: docs/snippets/gna/configure.py
# Copyright (C) 2022 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

#! [import]
import openvino as ov
#! [import]

from snippets import get_model

def main():
    model = get_model()

    core = ov.Core()
    if "GNA" not in core.available_devices:
        return 0

    # TODO: no GNA properties to replace strings
    #! [ov_gna_exec_mode_hw_with_sw_fback]
    core = ov.Core()
    compiled_model = core.compile_model(
        model, device_name="GNA", config={"GNA_DEVICE_MODE": "GNA_HW_WITH_SW_FBACK"}
    )
    #! [ov_gna_exec_mode_hw_with_sw_fback]

File: docs/snippets/gna/configure.cpp
// Copyright (C) 2022 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//
//! [include]
#include <openvino/openvino.hpp>
#include <openvino/runtime/intel_gna/properties.hpp>
//! [include]

int main() {
const std::string model_path = "model.xml";
//! [ov_gna_exec_mode_hw_with_sw_fback]
ov::Core core;
auto model = core.read_model(model_path);
auto compiled_model = core.compile_model(model, "GNA",
   ov::intel_gna::execution_mode(ov::intel_gna::ExecutionMode::HW_WITH_SW_FBACK));
//! [ov_gna_exec_mode_hw_with_sw_fback]
}